src/pages/iou/WaypointEditor.js

```diff
@@ -0,0 +1,149 @@
+import React from 'react';
+import {View} from 'react-native';
+import PropTypes from 'prop-types';
+import {withOnyx} from 'react-native-onyx';
+import AddressSearch from '../../components/AddressSearch';
+import ScreenWrapper from '../../components/ScreenWrapper';
+import HeaderWithBackButton from '../../components/HeaderWithBackButton';
+import Navigation from '../../libs/Navigation/Navigation';
+import ONYXKEYS from '../../ONYXKEYS';
+import Form from '../../components/Form';
+import styles from '../../styles/styles';
+import compose from '../../libs/compose';
+import CONST from '../../CONST';
+import withLocalize, {withLocalizePropTypes} from '../../components/withLocalize';
+import * as Transaction from '../../libs/actions/Transaction';
+import * as ValidationUtils from '../../libs/ValidationUtils';
+import ROUTES from '../../ROUTES';
+import {withNetwork} from '../../components/OnyxProvider';
+import networkPropTypes from '../../components/networkPropTypes';
+
+const propTypes = {
+    route: PropTypes.shape({
+        params: PropTypes.shape({
+            /** IOU type */
+            iouType: PropTypes.string,
+
+            /** Index of the waypoint being edited */
+            waypointIndex: PropTypes.string,
+
+            /** The transactionID of this request */
+            transactionID: PropTypes.string,
```

Comment:
Oh, I didn't think it was possible but it would work fine since you're loading the IOU onyx object with a transaction ID when you go to the distance tab.

The advantages are just that you aren't relying on the transactionID existing in the object, but I think it's cleaner without the transactionID in the URL so I removed it